### PR TITLE
add RevMap and String to enums

### PR DESCRIPTION
**Pull request by mlsteele**
Add `FooRevMap` for enums. Go only.
Just like #3 but backwards, for `Type -> String`.

r? @maxtaco

### CODE REVIEW

File: src/emit.iced [hunk 1]
```diff
@@ -305,6 +307,16 @@ exports.GoEmitter = class GoEmitter
     @untab()
     @output "}"
 
+    # Reverse map
+    @output "var #{name}RevMap = map[#{name}]string{"
+    @tab()
+    for s, i in t.symbols
+      i = s.split("_").pop(-1);
```
Comment (maxtaco) [inline src/emit.iced]: does `shift` work here instead?

File: src/emit.iced [hunk 2]
```diff
@@ -305,6 +307,16 @@ exports.GoEmitter = class GoEmitter
...
+      s = s.replace("_" + i, "");
```
Comment (maxtaco) [inline src/emit.iced]: I had to squint at this, maybe just `join` the stuff you didn't pop above?

**Comment by maxtaco** [comment]
small comments

**Comment by mlsteele** [comment]
`shift` might fail if the type name has an "_" in it. Hopefully that never happens, but I think this is still safer. Is this better?

**Comment by maxtaco** [comment]
Sorry I meant pop() without the argument?

Definitely think you should guard against the multiple _ case

**Comment by maxtaco** [comment]
Ok after those small changes! I will need to push to npm for you.

**Comment by maxtaco** [comment]
👍

**Comment by mlsteele** [comment]
Cool, thanks! Lmk when you push to npm and I'll update the client.

**Comment by maxtaco** [comment]
done

**Pull request merged**